nshlib/nsh_proccmds.c

```diff
@@ -801,6 +801,107 @@ int cmd_kill(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
 }
 #endif
 
+/****************************************************************************
+ * Name: cmd_pkill
+ ****************************************************************************/
+
+#if defined(CONFIG_FS_PROCFS) && !defined(CONFIG_NSH_DISABLE_PKILL)
+int cmd_pkill(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
+{
+  FAR const char *name;
+  FAR char *ptr;
+  pid_t pids[8];
+  long signal;
+  ssize_t ret;
+  int i;
+
+  /* pkill will send SIGTERM to the task in case no signal is selected by
+   * -<signal> option
+   */
+
+  if (argc == 3)  /* pkill -<signal> <name> */
+    {
+      /* Check incoming parameters.
+       * The first parameter should be "-<signal>"
+       */
+
+      ptr = argv[1];
+      if (*ptr != '-' || ptr[1] < '0' || ptr[1] > '9')
+        {
+          goto invalid_arg;
+        }
+
+      /* Extract the signal number */
+
+      signal = strtol(&ptr[1], NULL, 0);
```

Comment:
Shouldn't the signal range be checked before sending it to kill() ?

Reply:
`kill()` does the check itself and returns `EINVAL` if signal is not valid.